packages/nextra-theme-docs/src/components/sidebar.tsx

```diff
@@ -173,6 +237,12 @@ function FolderImpl({ item, anchors }: FolderProps): ReactElement {
           TreeState[item.route] = !open
           rerender({})
         }}
+        onFocus={() => {
+          focusControl?.setFocused(item.route)
+        }}
+        onBlur={() => {
+          focusControl?.setBlurred(item.route)
+        }}
```

Comment:
Nice catch and reactivity to fix this issue (I also discovered it yday 😀). I think simple omitting `onBlur` handler should fix this confusing behaviour when `autoColapse` is set to `true`

Reply:
My thought is that if we omit `onBlur`, the collapse may not automatically close when keyboard navigation shifts focus away from a sidebar item, such as switching to the locale switch.

Reply:
I think it's fine to have this case, instead of introducing complexity with `setTimeout`s IMO

Reply:
Yes, I agree the setTimeout approach isn't ideal. I will fix it in the next commit.

Reply:
I think we can remove the `onBlur` prop (regardless of whether `sidebar.autoCollapse` is enabled).

By using the sidebar's `focusout` event (currently the DOM referenced in `sidebarRef`), the use of `setTimeout` becomes meaningful, as we only need to check if `document.activeElement` is still within the sidebar during the most recent `focusout` in sidebar.

This logic is consolidated within a single `useEffect`, which also reduces the re-renders triggered by the `onBlur` when calling `setFocused(null)`. Does this approach look good to you?

```ts
useEffect(() => {
  const sidebarEl = sidebarRef.current;
  if (!sidebarEl) return;
  let timer: ReturnType<typeof setTimeout>;

  const handleFocusout = () => {
    clearTimeout(timer);
    timer = setTimeout(() => {
      if (sidebarEl?.contains(document.activeElement)) return;
      setFocused(null);
    }, 50);
  };

  sidebarEl.addEventListener('focusout', handleFocusout);
  return () => {
    clearTimeout(timer);
    sidebarEl.removeEventListener('focusout', handleFocusout);
  };
}, []);
```